### ISSUE TITLE
Sweeping Strikes

### ISSUE THREAD

**Issue by kaleharbick** [magey/tbc-warrior#7]
Verify all of this still applies:

- sweeps can crit but do a crit roll based on spell crit
- sweeps calculate dmg based on primary dmg (no reroll of wep dmg is done for example) but using secondary's armor
- if primary target is crit, dmg will match on secondary (if armor matches) but secondary hit won't be an actual "crit"
- cleave uses 2 charges of ss (4 hits go out, 2x cleave and 2x ss)
- ww uses 4 charges of ss (8 hits go out, 4x ww and 4x ss)
- execute on a primary causes execute on secondary if secondary is <20%, otherwise causes white hit on secondary

**Comment by gillismr** [comment]
Khalorl uploaded a video of some SS+WW and SS+Cleave interactions. They are using a 2 handed weapon: 
https://discordapp.com/channels/383596811517952002/784258037187739648/824385944181276673

Cleave appears to be working properly. 2 targets, 2 charges, 4 yellow hits: 
https://cdn.discordapp.com/attachments/784258037187739648/824390819124871229/unknown.png

WW appears to be bugged. 3 targets, 3 yellow hits. Not sure if an SS charge is consumed due to limited combat log.
https://cdn.discordapp.com/attachments/784258037187739648/824390819124871229/unknown.png

Also, in MonkeyNew's stream, you can see countless examples, although it's a little less clear because he's dual wielding. 
https://www.twitch.tv/videos/960275747?t=07h19m00s
Right at 07:19:06, SS is activated, one charge is used, hitting 2 vultures for 219. Then WW is used and it appears that only 6 new yellow hits pop up (excluding a deep wounds tick for <100 dmg) for 3 targets. The next two white hits to go out are for 658 and 227, so if you see those, don't include them in your WW hit-count. I see 6 yellow hits from WW, possibly one SS charge consumed but no additional hits. Seems broken.

**Comment by magey** [comment]
So, first of all some research I've done regarding SS+WW interaction while dual-wielding in original TBC:

[Video 1](https://www.youtube.com/watch?v=bWYyXMgmeFU)
In this video the warrior is attacking 3 targets (things happen quickly so play this at x0.25 rate). He casts SS and swings with both weapons. One weapon hits for 185 and the other misses (Rate is French for Missed), followed by a yellow 185 hit from SS. He then casts WW immediately followed by Cleave. WW generates 7 yellow hits (182, 356, 454, 398, 779, 794 and Esquive which is French for Dodge); 6 from WW itself and one more from SS.

[Video 2](https://www.youtube.com/watch?v=Waq5Hgg7rZ4&t=28s)
Here we can see that WW+SS did not generate even 1 SS hit, probably because he was moving as he cast WW and due to SS having a short range and a 180 degree frontal arc the selected SS target was not in range. If WW actually generated 8 SS hits one of them would have hit one of the mobs.

Now, on beta the behavior is identical to the videos above:

While dual wielding:
![image](https://user-images.githubusercontent.com/44882404/112698196-35950b00-8e9a-11eb-8136-fe8dc0da0282.png)

While using a 2H:
![image](https://user-images.githubusercontent.com/44882404/112698210-3ded4600-8e9a-11eb-85e3-6aef42d97ac7.png)

In conclusion, SS seems to be working correctly on the beta (for now).
